homeassistant/components/melnor/sensor.py

```diff
@@ -0,0 +1,133 @@
+"""Support for Melnor RainCloud sprinkler water timer."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from melnor_bluetooth.device import Device
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, SIGNAL_STRENGTH_DECIBELS_MILLIWATT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .models import MelnorBluetoothBaseEntity, MelnorDataUpdateCoordinator
+
+
+@dataclass
+class MelnorSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    state_fn: Callable[[Device], Any]
+
+
+@dataclass
+class MelnorSensorEntityDescription(
+    SensorEntityDescription, MelnorSensorEntityDescriptionMixin
+):
+    """Describes Melnor sensor entity."""
+
+    icon_fn: Callable[[Device], str] | None = None
+
+
+def _rssi_icon(device: Device) -> str:
+    """Return a representative icon based on signal strength."""
+
+    if device.rssi >= -20:
+        return "mdi:wifi-strength-4"
+    if device.rssi >= -40:
+        return "mdi:wifi-strength-3"
+    if device.rssi >= -60:
+        return "mdi:wifi-strength-2"
+    if device.rssi >= -80:
+        return "mdi:wifi-strength-1"
+    return "mdi:wifi-strength-outline"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+
+    coordinator: MelnorDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors: list[MelnorSensorEntityDescription] = [
+        MelnorSensorEntityDescription(
+            device_class=SensorDeviceClass.BATTERY,
+            entity_category=EntityCategory.DIAGNOSTIC,
+            has_entity_name=True,
```

Comment:
`has_entity_name` is redundant, as it is already set via `_attr_has_entity_name`:
```suggestion
```